src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,2068 @@
+              params: [],
+              skipCache: true,
+            },
+            expect.any(Function),
+          );
+        },
+      );
+    });
+
+    it('should not cause the block tracker to make a request for the latest block number more than once when called more than once while the current block number cache is still fresh', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ provider, blockTracker }) => {
+        const sendAsyncSpy = jest.spyOn(provider, 'sendAsync');
+        await blockTracker.getLatestBlock();
+        await blockTracker.getLatestBlock();
+        const requestsForLatestBlock = sendAsyncSpy.mock.calls.filter(
+          (args) => {
+            return args[0].method === 'eth_blockNumber';
+          },
+        );
+        expect(requestsForLatestBlock).toHaveLength(1);
+      });
+    });
+
+    it('should cause the block tracker to make a request for the latest block number again if called again after the current block number cache becomes stale', async () => {
+      const setTimeoutRecorder = recordCallsToSetTimeout();
+      const blockTrackerOptions = {
+        pollingInterval: 100,
+        blockResetDuration: 200,
+      };
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x0',
+                },
+              },
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x1',
+                },
+              },
+            ],
+          },
+          blockTracker: blockTrackerOptions,
+        },
+        async ({ provider, blockTracker }) => {
+          const sendAsyncSpy = jest.spyOn(provider, 'sendAsync');
+          await blockTracker.getLatestBlock();
+          // When the block tracker stops, there may be two `setTimeout`s in
+          // play: one to go to the next iteration of the block tracker
+          // loop, another to expire the current block number cache. We don't
+          // know which one has been added first, so we have to find it.
+          await setTimeoutRecorder.nextMatchingDuration(
+            blockTrackerOptions.blockResetDuration,
+          );
+          await blockTracker.getLatestBlock();
+          const requestsForLatestBlock = sendAsyncSpy.mock.calls.filter(
+            (args) => {
+              return args[0].method === 'eth_blockNumber';
+            },
+          );
+          expect(requestsForLatestBlock).toHaveLength(2);
+        },
+      );
+    });
+
+    METHODS_TO_ADD_LISTENER.forEach((methodToAddListener) => {
+      it(`should emit the "error" event (added via \`${methodToAddListener}\`) and should not throw if, while making the request for the latest block number, the provider throws an error that is an Error`, async () => {
```

Comment:
Hmm. What's the relevance of the type of the error? e.g. string v.s `Error`

Reply:
This is to test the `err.stack ?? err` logic [here](https://github.com/MetaMask/eth-block-tracker/blob/e4e4e27d74ca83cccb2a93acea20478d55db5144/src/PollingBlockTracker.ts#L69). Strings don't have a `stack` property.